python/private/local_runtime_repo_setup.bzl

```diff
@@ -69,44 +79,70 @@ def define_local_runtime_toolchain_impl(
     # See https://docs.python.org/3/extending/windows.html
     # However not all python installations (such as manylinux) include shared or static libraries,
     # so only create the import library when interface_library is set.
-    full_abi_deps = []
-    abi3_deps = []
     if interface_library:
         cc_import(
-            name = "_python_interface_library",
+            name = "interface",
             interface_library = interface_library,
-            system_provided = 1,
+            system_provided = True,
         )
-        if interface_library.endswith("{}.lib".format(major)):
-            abi3_deps = [":_python_interface_library"]
-        else:
-            full_abi_deps = [":_python_interface_library"]
 
-    cc_library(
-        name = "_python_headers_abi3",
-        # NOTE: Keep in sync with watch_tree() called in local_runtime_repo
+    if abi3_interface_library:
+        cc_import(
+            name = "abi3_interface",
+            interface_library = abi3_interface_library,
+            system_provided = True,
+        )
+
+    native.filegroup(
+        name = "includes",
         srcs = native.glob(
             include = ["include/**/*.h"],
             exclude = ["include/numpy/**"],  # numpy headers are handled separately
             allow_empty = True,  # A Python install may not have C headers
         ),
-        deps = abi3_deps,
+    )
+
+    # header libraries.
+    cc_library(
+        name = "python_headers_abi3",
+        hdrs = [":includes"],
+        includes = ["include"],
+        defines = defines,  # NOTE: Users should define Py_LIMITED_API=3
+        deps = select({
+            "@bazel_tools//src/conditions:windows": [":abi3_interface"],
+            "//conditions:default": [],
+        }),
+    )
+
+    cc_library(
+        name = "python_headers",
+        hdrs = [":includes"],
         includes = ["include"],
+        defines = defines,
+        deps = select({
+            "@bazel_tools//src/conditions:windows": [":interface"],
+            "//conditions:default": [],
+        }),
     )
+
+    # python libraries
     cc_library(
-        name = "_python_headers",
-        deps = [":_python_headers_abi3"] + full_abi_deps,
+        name = "libpython_abi3",
+        hdrs = [":includes"],
+        defines = defines,  # NOTE: Users should define Py_LIMITED_API=3
+        srcs = abi3_libraries + additional_dlls,
     )
 
     cc_library(
-        name = "_libpython",
-        hdrs = [":_python_headers"],
-        srcs = libraries,
-        deps = [],
+        name = "libpython",
+        hdrs = [":includes"],
+        defines = defines,
+        srcs = libraries + additional_dlls,
     )
 
+    # runtime configuration
     py_runtime(
-        name = "_py3_runtime",
+        name = "py3_runtime",
```

Comment:
is there a particular reason these various targets were made public?

Reply:
They were changed to mimic the hermetic toolchain names. I'm not sure whether there is a benefit to one vs. the other.